addons/im_livechat/__manifest__.py

```diff
@@ -79,153 +75,77 @@
         ],
         'web.qunit_suite_tests': [
             'im_livechat/static/tests/**/*',
+            ('remove', 'im_livechat/static/tests/embed/**/*'),
             ('remove', 'im_livechat/static/tests/tours/**/*'),
             ('remove', 'im_livechat/static/tests/helpers/**/*.js'),
         ],
         'web.assets_tests': [
             'im_livechat/static/tests/tours/**/*',
         ],
-        'im_livechat.assets_public_livechat': [
-            'im_livechat/static/src/legacy/model.js',
-            'im_livechat/static/src/legacy/model/*',
-            'im_livechat/static/src/legacy/core_models/*',
-            'im_livechat/static/src/livechat_data.js',
-            'im_livechat/static/src/legacy/legacy_models/*',
-            'im_livechat/static/src/legacy/widgets/*',
-            'im_livechat/static/src/legacy/widgets/*/*',
-            'im_livechat/static/src/legacy/public_models/*.js',
+        'im_livechat.shared_frontend_public_assets': [
+            'mail/static/src/core/**/*',
+            'mail/static/src/core_ui/**/*',
+            'mail/static/src/utils/**/*',
+            'mail/static/src/emoji_picker/**/*',
+            ('remove', 'mail/static/src/emoji_picker/emoji_data.js'),
+            'mail/static/src/attachments/**/*',
+            'mail/static/src/composer/**/*',
+            'mail/static/src/dropzone/**/*',
+            'mail/static/src/rtc**/*',
+            'mail/static/src/chat_window/**/*',
+            'mail/static/src/web/discuss_app/context_service.js',
```

Comment:
Could move the `mail/` file in `core/`, so that you don't have to explicitly import here.
The public page could explicitly remove it in manifest and add its own

Reply:
Not sure about this. I would make the diff even heavier. 
- Move of the files
- Rename of all imports
- Changes in mail manifest for public page
IMO it's not a big deal to keep it as is. What do you think?